src/components/car/car_form.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+import './car_form.css';
+
+interface Car {
+  id: number;
+  maker: string;
+  model: string;
+  year: number;
+}
+
+const initialCarState: Car = {
+  id: 0,
+  maker: 'Toyota',
+  model: 'Toyota RQ',
+  year: 2000,
+};
+
+// interface CarFormProps {
+//   cars: Car[];
+//   setCars: (cars: Car[]) => void;
+//   addCar: (car: Car) => void;
+// }
+
+export default function CarForm({ addCar }: { addCar: (car: Car) => void }) {
+  const [car, setCar] = useState<Car>(initialCarState); // Car object state
+  const [errors, setErrors] = useState<string[]>([]); // Errors state
+
+  const validate = () => {
+    const newErrors: string[] = [];
+    if (!car.maker) {
+      newErrors.push('Make is required');
+    }
+    if (!car.model) {
+      newErrors.push('Model is required');
+    }
+    if (car.year <= 1900) {
+      newErrors.push('Year must be greater than 1900');
+    }
+    setErrors(newErrors);
+    return newErrors.length === 0;
+  };
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    setCar({ ...car, [name]: value });
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+    setCar({ ...car, maker: value });
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (validate()) {
+      addCar(car);
+      setCar(initialCarState);
+    }
```

Comment:
For a better user experience, it would be nice to clean the form fields after user successfully submit it. You can improve it later.

Reply:
the form fields are cleaned after successfully submit